File: backend-v5/lms/poll/Poll.php
<?php

// Modul: lms/poll
// Klasa: Poll
// Opis: jedna anketa

require_once(Config::$backend_path."core/DB.php");

class Poll {
	public $id;
	public $name, $description, $openDate, $closeDate, $editable;
	
	public static function fromId($id) {
		$q10 = DB::query("select naziv, opis, UNIX_TIMESTAMP(datum_otvaranja), UNIX_TIMESTAMP(datum_zatvaranja), editable from anketa_anketa where id=$id");
		if (mysql_num_rows($q10)<1) {
			throw new Exception("unknown poll");
		}
		
		$p = new Poll;
		$p->id = $id;
		$p->name = mysql_result($q10,0,0);
		$p->description = mysql_result($q10,0,1);
		$p->openDate = mysql_result($q10,0,2);
		$p->closeDate = mysql_result($q10,0,3);
		if (mysql_result($q10,0,4) == 1) $p->editable=true; else $p->editable=false;
		
		return $p;
	}
	
	// Gets just one poll that is active for all courses in current academic year 
	// (usually there aren't more such polls)
	public static function getActiveForAllCourses() {
		$q10 = DB::query("select aa.id, aa.naziv, aa.opis, UNIX_TIMESTAMP(aa.datum_otvaranja), UNIX_TIMESTAMP(aa.datum_zatvaranja), aa.editable from anketa_anketa as aa, akademska_godina as ag, anketa_predmet as ap where where aa.id=ap.anketa and ap.aktivna=1 and ap.predmet=0 and aa.akademska_godina=ag.id and ag.aktuelna=1 order by id desc limit 1");
		if (mysql_num_rows($q10)<1) {
			throw new Exception("no currently active poll");
		}
		
		$p = new Poll;
		$p->id = mysql_result($q10,0,0);
		$p->name = mysql_result($q10,0,1);
		$p->description = mysql_result($q10,0,2);
		$p->openDate = mysql_result($q10,0,3);
		$p->closeDate = mysql_result($q10,0,4);
		if (mysql_result($q10,0,5) == 1) $p->editable=true; else $p->editable=false;
		
		return $p;
	}

	// Gets just one poll that is active for given course and given academic year
	public static function getActiveForCourse($courseUnitId, $academicYearid) {
		$q10 = DB::query("select aa.id, aa.naziv, aa.opis, UNIX_TIMESTAMP(aa.datum_otvaranja), UNIX_TIMESTAMP(aa.datum_zatvaranja), aa.editable from anketa_anketa as aa, anketa_predmet as ap where where aa.id=ap.anketa and ap.aktivna=1 and ((ap.predmet=$courseUnitId and ap.akademska_godina=$academicYearid) or ap.predmet=0) order by aa.id desc limit 1");
		if (mysql_num_rows($q10)<1) {
			throw new Exception("no currently active poll");
		}
		
		$p = new Poll;
		$p->id = mysql_result($q10,0,0);
		$p->name = mysql_result($q10,0,1);
		$p->description = mysql_result($q10,0,2);
		$p->openDate = mysql_result($q10,0,3);
		$p->closeDate = mysql_result($q10,0,4);
		if (mysql_result($q10,0,5) == 1) $p->editable=true; else $p->editable=false;
		
		return $p;
	}
	
	// Returns true if poll is active for all courses OR for given course, otherwise returns false
	public function isActiveForCourse($courseUnitId, $academicYearId) {
		$q10 = DB::query("select aktivna from anketa_predmet where predmet=0 and anketa=".$this->id);
		if (mysql_num_rows($q10)<1 || mysql_result($q10,0,0)!=1) {
			$q20 = DB::query("select aktivna from anketa_predmet where predmet=$courseUnitId and akademska_godina=$academicYearId and anketa=".$this->id);
			if (mysql_num_rows($q20)<1 || mysql_result($q20,0,0)!=1) return false;
		}
		return true;
	}
}

?>

File: rorbackend/app/models/lms/poll/poll.rb
class Lms::Poll::Poll < ActiveRecord::Base
  # Uncomment following lines if working with legacy database
  # set_table_name 'anketa_anketa'
  # set_primary_key :id
  # alias_attribute :id, :id
  # alias_attribute :open_date, :datum_otvaranja
  # alias_attribute :close_date, :datum_zatvaranja
  # alias_attribute :name, :naziv
  # alias_attribute :description, :opis
  # alias_attribute :active, :aktivna
  # alias_attribute :editable, :editable
  # alias_attribute :academic_year_id, ;akademska_godina

  # Uncomment following lines if working with legacy database
  # TABLE_NAME = 'anketa_anketa'
  # ID = TABLE_NAME + '.' + 'id'
  # OPEN_DATE = TABLE_NAME + '.' + 'datum_otvaranja'
  # CLOSE_DATE = TABLE_NAME + '.' + 'datum_zatvaranja'
  # NAME = TABLE_NAME + '.' + 'naziv'
  # DESCRIPTION = TABLE_NAME + '.' + 'opis'
  # ACTIVE = TABLE_NAME + '.' + 'aktivna'
  # EDITABLE = TABLE_NAME + '.' + 'editable'
  # ACADEMIC_YEAR_ID = TABLE_NAME + '.' + 'akademska_godina'

  # Comment following lines if working with legacy database
  TABLE_NAME = 'lms_poll_polls'
  ID = TABLE_NAME + '.' + 'id'
  OPEN_DATE = TABLE_NAME + '.' + 'open_date'
  CLOSE_DATE = TABLE_NAME + '.' + 'close_date'
  NAME = TABLE_NAME + '.' + 'name'
  DESCRIPTION = TABLE_NAME + '.' + 'description'
  ACTIVE = TABLE_NAME + '.' + 'active'
  EDITABLE = TABLE_NAME + '.' + 'editable'
  ACADEMIC_YEAR_ID = TABLE_NAME + '.' + 'academic_year_id'

  ALL_COLUMNS = [ID, OPEN_DATE, CLOSE_DATE, NAME, DESCRIPTION, ACTIVE, EDITABLE, ACADEMIC_YEAR_ID]
  
  belongs_to :academic_year, :class_name => "Core::AcademicYear"
end
